### PR TITLE
chore: update shell.nix and flake.lock tooling, documentation

### PR DESCRIPTION
**Pull request by andrevmatos**
This should work with current version of nixpkgs repositories. Could not test integration tests, but basic development dependencies and tests are working on this version of `shell.nix`.

### CODE REVIEW

**Review by connorwstein — APPROVED** [review]
LGTM, but might be better if @archseer can take a look

**Review by connorwstein — APPROVED** [review]

File: shell.nix [hunk 1]
```diff
@@ -1,34 +1,50 @@
...
-  LD_LIBRARY_PATH="${stdenv.cc.cc.lib}/lib64:$LD_LIBRARY_PATH";
-  GOROOT="${pkgs.go_1_17}/share/go";
+  LD_LIBRARY_PATH = "${stdenv.cc.cc.lib}/lib64:$LD_LIBRARY_PATH";
+  GOROOT = "${go}/share/go";
 
-  PGDATA="db";
+  PGDATA = "db";
+  DATABASE_URL = "postgresql://chainlink:chainlink@localhost:5432/chainlink_test?sslmode=disable";
+  shellHook = ''
+    export GOPATH=$HOME/go
```
Comment (archseer) [inline shell.nix:46]: Hmm, is this required? Could maybe use `export GOPATH=$(${go}/bin/go env GOPATH)`

**Review by archseer — APPROVED** [review]

File: shell.nix (hunk 1 above)
Comment (archseer) [reply shell.nix:46]: Actually, both `GOROOT` and this `shellHook` should be removed, they're not needed anymore

**Review by connorwstein — APPROVED** [review]

**Comment by andrevmatos** [comment]
Replaced by #7715

**Pull request closed without merging**